### PR TITLE
Separate features between Admin Portal & Developer Portal

### PR DESCRIPTION
**Pull request by brionmario**
## Purpose
Handling admin tasks inside the Developer Portal is not ideal and it would be much better if there is a separate portal to carry out those responsibilities.

## Goals

This is a continuation of #938. With this PR, a clear separation of features in Developer Portal and Admin Portal will be introduced.

Following features will be available in respective portals.

**Developer Portal**
- Overview
- Applications
- Identity Providers

**Admin Portal**
- Overview
- Users
- Groups
- Roles
- Attribute Dialects
- Userstores
- Certificates
- Configurations
    - General
    - Email Templates

## Approach

Re-usable logic was separated out to `@wso2is/core` module.

`@wso2is/core/api`
1. **claims.ts** - Claim replated common API calls.
2. **roles.ts** - Role replated common API calls.
3. **userstores.ts** - Userstore replated common API calls.

`@wso2is/core/constants`
1. **certificate-management-constants.ts** - Constants used for certificate related operation.
2. **claim-constants.ts** - Claim management related constants.
3. **userstore-constants.ts** - Userstore replated constants.

`@wso2is/core/models`
1. **certificates.ts** - Certificate common interfaces and Enums.
2. **claim.ts** - Claim management common interfaces and Enums.
3. **common.ts** - Common utility interfaces.
4. **roles.ts** - Role management common interfaces and Enums.
5. **userstores.ts** - Userstore management common interfaces and Enums.

`@wso2is/core/utils`

1. **certificate-management-utils.ts.ts** - Certificate encode, decode etc. utils.
2. **common-utils.ts.ts** - Commonly used useful utils.

Re-usable components were separated out to `@wso2is/react-components` module.

1. `AnimatedAvatar` - Moved and renamed `AvatarBackground` component.
2. `DynamicField` - Moved the original component.
3. `FileUpload` - Moved the and renamed the original `UploadFile` component.
4. `URLInputComponent ` - Moved the and renamed the original `URLInput` component.
4. `Certificate ` - Moved the original component.

## TODO
1. Port the SBAC refactoring to developer portal as well.
2. Fix Developer Portal & Admin Portal overview pages.

### CODE REVIEW

File: apps/admin-portal/src/components/shared/advanced-search-with-basic-filters.tsx [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { TestableComponentInterface } from "@wso2is/core/models";
+import { SearchUtils } from "@wso2is/core/utils";
+import { DropdownChild, Field, Forms } from "@wso2is/forms";
+import { AdvancedSearch, LinkButton, PrimaryButton } from "@wso2is/react-components";
+import { AdvancedSearchPropsInterface } from "@wso2is/react-components/src";
```
Comment (NipuniBhagya) [inline apps/admin-portal/src/components/shared/advanced-search-with-basic-filters.tsx:23]: Shall we import this from react-components?

File: apps/admin-portal/src/components/user-stores/user-stores-list.tsx [hunk 2]
```diff
@@ -17,7 +17,7 @@
 */
 
 import { hasRequiredScopes } from "@wso2is/core/helpers";
-import { LoadableComponentInterface, SBACInterface, TestableComponentInterface } from "@wso2is/core/models";
+import { AlertLevels, LoadableComponentInterface, SBACInterface, TestableComponentInterface } from "@wso2is/core/models";
```
Comment (NipuniBhagya) [inline apps/admin-portal/src/components/user-stores/user-stores-list.tsx:20]: Possibly exceeding the maximum length.

File: apps/developer-portal/src/app.tsx [hunk 3]
```diff
@@ -29,17 +29,17 @@ import { LocalStorageUtils } from "@wso2is/core/utils";
...
 import { useDispatch, useSelector } from "react-redux";
 import { Redirect, Route, Router, Switch } from "react-router-dom";
 import { ProtectedRoute } from "./components";
 import { Config, baseRoutes } from "./configs";
-import { ApplicationConstants } from "./constants";
+import { AppConstants } from "./constants";
 import { history } from "./helpers";
 import {
-    ConfigReducerStateInterface,
+    ConfigReducerStateInterface, DeploymentConfigInterface,
```
Comment (NipuniBhagya) [inline apps/developer-portal/src/app.tsx:42]:
```suggestion
    ConfigReducerStateInterface, 
    DeploymentConfigInterface,
```

File: apps/developer-portal/src/components/protected-route.tsx [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (NipuniBhagya) [inline apps/developer-portal/src/components/protected-route.tsx:2]:
```suggestion
 * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: apps/developer-portal/src/index.tsx [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (NipuniBhagya) [inline apps/developer-portal/src/index.tsx:2]:
```suggestion
 * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: apps/developer-portal/src/layouts/default.tsx [hunk 6]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (NipuniBhagya) [inline apps/developer-portal/src/layouts/default.tsx:2]:
```suggestion
 * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: modules/core/src/models/userstores.ts [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    description: string;
+    /**
+     * Location of the created/updated resource.
+     * example: /t/{tenant-domain}/api/server/v1/userstores/SkRCQy1TRUNPTkRBUlk
+     */
+    self: string;
+    /**
+     * Requested configured user store property for the set.
+     */
+    properties:	UserstorePropertiesResponseInterface[];
```
Comment (NipuniBhagya) [inline modules/core/src/models/userstores.ts:43]:
```suggestion
    properties: UserstorePropertiesResponseInterface[];
```

Comment (brionmario) [reply modules/core/src/models/userstores.ts]: Fixed with commit `#0c51a66`.

File: apps/developer-portal/src/index.tsx (hunk 5 above)
Comment (brionmario) [reply apps/developer-portal/src/index.tsx]: Fixed with commit `#0c51a66`.

File: apps/admin-portal/src/api/users.ts [hunk 8]
```diff
@@ -192,3 +192,59 @@ export const updateUserRoles = (data: object): Promise<any> => {
...
+        })
+        .catch((error) => {
+            return Promise.reject(`Failed to retrieve user information - ${error}`);
+        });
+};
+
+/**
+ * Update the required details of the user profile.
+ *
+
```
[the remark this answers is not in the document]
Comment (brionmario) [reply apps/admin-portal/src/api/users.ts]: Fixed with commit `#0c51a66`.

File: apps/admin-portal/src/components/claims/claims-list.tsx [hunk 9]
```diff
@@ -17,29 +17,39 @@
 */
 
 import { hasRequiredScopes } from "@wso2is/core/helpers";
-import { LoadableComponentInterface, SBACInterface, TestableComponentInterface } from "@wso2is/core/models";
+import {     AlertLevels,
```
[the remark this answers is not in the document]
Comment (brionmario) [reply apps/admin-portal/src/components/claims/claims-list.tsx]: Fixed with commit `#0c51a66`.

File: apps/admin-portal/src/components/shared/advanced-search-with-basic-filters.tsx (hunk 1 above)
Comment (brionmario) [reply apps/admin-portal/src/components/shared/advanced-search-with-basic-filters.tsx]: Fixed with commit `#0c51a66`.

File: apps/admin-portal/src/components/user-stores/user-stores-list.tsx (hunk 2 above)
Comment (brionmario) [reply apps/admin-portal/src/components/user-stores/user-stores-list.tsx]: Fixed with commit `#0c51a66`.

File: apps/developer-portal/src/app.tsx (hunk 3 above)
Comment (brionmario) [reply apps/developer-portal/src/app.tsx]: Fixed with commit `#0c51a66`.

File: apps/developer-portal/src/layouts/default.tsx (hunk 6 above)
Comment (brionmario) [reply apps/developer-portal/src/layouts/default.tsx]: Fixed with commit `#61eff80`.

**Pull request merged**